openpathsampling/volume.py

```diff
@@ -10,12 +10,24 @@
 # TODO: Make Full and Empty be Singletons to avoid storing them several times!
 
 def join_volumes(volume_list):
-    volume = None
+    """
+    Make the union of a list of volumes. (Useful shortcut.)
+
+    Parameters
+    ----------
+    volume_list : list of Volume
+        the list to be joined together
+
+    Returns
+    -------
+    UnionVolume 
+        the union of the elements of the list, or EmptyVolume if list is
+        empty
+    """
+    volume = EmptyVolume()
+    # EmptyVolume is smart and knows its OR just takes the other
     for vol in volume_list:
-        if volume is None:
-            volume = vol
-        else:
-            volume = volume | vol
+        volume = volume | vol
```

Comment:
That would be better. I've been using `reduce` in a lot of other circumstances, but I don't usually think in terms of it. I find it harder to read (I almost always add a comment to explain what I'm doing with it), but in this specific case, the function is then just a clearly-named wrapper for a commonly used one-liner.

I won't worry about changing it in any immediate PR (just not a high priority), but I am in favor of the change.